tests/test_samples.py

```diff
@@ -0,0 +1,283 @@
+import json
+import os
+import re
+
+from tempfile import NamedTemporaryFile
+from mock import patch
+from tests.test_base import BaseClientTest
+from tests.test_value_constants import *
+from tests.mock_tieserver import MockTieServer
+
+if sys.version_info[0] > 2:
+    import builtins  # pylint: disable=import-error, unused-import
+    from urllib.parse import quote_plus  # pylint: disable=no-name-in-module, import-error, unused-import
```

Comment:
I think this and the `quote_plus` import in the `else` block below were both copied over from code in some of the other projects. I can't remember how that line was added in the first place since `quote_plus` didn't appear to be used in any the code in the tests. I went back and removed it from the ePO client, MISP client/service, and TheHive projects and the tests were still passing in Travis. I think, then, that it would be safe to drop these imports from this PR as well.

Reply:
Removed, thanks for the heads up. As a part of this, I'm also moving all of the sample running classes/functions into the test_base, so that it's easier to add to each project